spec/lib/vanagon/platform/windows_spec.rb

```diff
@@ -38,6 +57,106 @@
+          cur_plat._platform.generate_msi_packaging_artifacts(workdir,plat[:projname], binding)
+          # check the result
+          expect(FileUtils.compare_file("#{wixtestfiles}/file-3.wxs.erb", "#{workdir}/wix/file-wix.wxs")).to be_truthy
+        end
+
+        it "Pick Project Specific ERB File in favour of Generic ERB file" do
+          # setup source directories and run artifact generation
+          FileUtils.cp("#{wixtestfiles}/file-3.wxs.erb", "#{PROJ_ROOT}/resources/windows/wix/file-wix.wxs.erb")
+          FileUtils.cp("#{wixtestfiles}/file-4.wxs.erb", "#{VANAGON_ROOT}/resources/windows/wix/file-wix.wxs.erb")
+          cur_plat._platform.generate_msi_packaging_artifacts(workdir,plat[:projname], binding)
+          # check the result
+          expect(FileUtils.compare_file("#{wixtestfiles}/file-3.wxs.erb", "#{workdir}/wix/file-wix.wxs")).to be_truthy
+        end
+
+        it "copies Hierarchy of files from product specific directory to output directory" do
+          # setup source directories and run artifact generation
+          FileUtils.cp_r("#{wixtestfiles}/", "#{PROJ_ROOT}/resources/windows/", :verbose => true)
+          cur_plat._platform.generate_msi_packaging_artifacts(workdir,plat[:projname], binding)
+          # check the result
+          expect(File).to exist("#{workdir}/wix/file-1.wxs")
+          expect(File).to exist("#{workdir}/wix/file-2.wxs")
+          expect(File).to exist("#{workdir}/wix/file-3.wxs")
+          expect(File).to exist("#{workdir}/wix/file-4.wxs")
+          expect(File).to exist("#{workdir}/wix/project.filter.xslt")
+          expect(File).to exist("#{workdir}/wix/project.wxs")
+          expect(File).to exist("#{workdir}/wix/include/include-sample-1.wxs")
+          expect(File).to exist("#{workdir}/wix/ui/ui-sample-1.wxs")
+          expect(File).to exist("#{workdir}/wix/ui/bitmaps/bitmap.bmp")
+        end
+
+        it "copies Hierarchy of files from vanagon directory to output directory" do
+          # setup source directories and run artifact generation
+          FileUtils.cp_r("#{wixtestfiles}/", "#{VANAGON_ROOT}/resources/windows/", :verbose => true)
+          cur_plat._platform.generate_msi_packaging_artifacts(workdir,plat[:projname], binding)
+          # check the result
+          expect(File).to exist("#{workdir}/wix/file-1.wxs")
+          expect(File).to exist("#{workdir}/wix/file-2.wxs")
+          expect(File).to exist("#{workdir}/wix/file-3.wxs")
+          expect(File).to exist("#{workdir}/wix/file-4.wxs")
+          expect(File).to exist("#{workdir}/wix/project.filter.xslt")
+          expect(File).to exist("#{workdir}/wix/project.wxs")
+          expect(File).to exist("#{workdir}/wix/include/include-sample-1.wxs")
+          expect(File).to exist("#{workdir}/wix/ui/ui-sample-1.wxs")
+          expect(File).to exist("#{workdir}/wix/ui/bitmaps/bitmap.bmp")
+        end
+
+        after do
```

Comment:
same as before; additionally, put all setup/teardown code at the top of the example group. People shouldn't need to go hunting for essential pieces like this.